security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/ActionDeserializer.java

```diff
@@ -122,6 +151,24 @@ public Action deserialize(JsonParser parser, DeserializationContext deserializat
                 } else {
                     return new PhaseTapChangerTapPositionAction(context.id, context.transformerId, context.relativeValue, context.value);
                 }
+            case GeneratorAction.NAME:
+                GeneratorAction generatorAction = new GeneratorAction(context.id, context.generatorId);
+                if (context.activePowerRelativeValue != null) {
+                    generatorAction.setActivePowerRelativeValue(context.activePowerRelativeValue);
+                }
+                if (context.activePowerValue != null) {
+                    generatorAction.setActivePowerValue(context.activePowerValue);
+                }
```

Comment:
both or none should be provided
```suggestion
                if ((context.activePowerRelativeValue != null) && (context.activePowerValue != null)) {
                    generatorAction.set... //TODO see other suggested changes
                } else if ((context.activePowerRelativeValue != null) ^ (context.activePowerValue != null)) {
                throw JsonMappingException.from(parser, "for generator action, both or none of these two attributes must be provided: activePowerValue, activePowerRelative");
                }
```